src/components/FilePicker.vue

```diff
@@ -77,97 +101,204 @@ export default {
-      '{DAV:}getlastmodified',
-      '{DAV:}getetag',
-      '{DAV:}resourcetype'
-    ],
-    isInitial: true
-  }),
-
-  computed: {
-    isSelectBtnEnabled() {
-      return this.isLocationPicker || this.areResourcesSelected
-    },
+  emits: ['select', 'cancel', 'update', 'folderLoaded'],
 
-    isLocationPicker() {
-      return this.variation === 'location'
-    },
+  setup(props, { emit }) {
+    let currentSpace = null
+    let isInitial = true
+
+    const { proxy } = getCurrentInstance() || {}
+    const { focusAndAnnounceBreadcrumb } = useA11y()
+
+    const client = inject<Ref<ReturnType<typeof webClient>>>('client')
+    const webdav = inject<Ref<ReturnType<typeof initWebdav>>>('webdav')
+    const sdk = inject<Ref<OwnCloudSdk>>('sdk')
+    const config = inject<Ref<FilePickerConfig>>('config')
+    const user = inject<Ref<User>>('user')
+    const capabilities = inject<Ref<Capabilities>>('capabilities')
+
+    const state = ref('loading')
+    const resources = ref([])
+    const currentFolder = ref(null)
+    const spaces = ref<Space[]>([])
+    const drives = ref<SpaceResource[]>([])
+    const selectedResources = ref([])
+
+    const isLocationPicker = computed(() => props.variation === 'location')
 
-    areResourcesSelected() {
-      return this.selectedResources.length > 0
+    const areResourcesSelected = computed(() => selectedResources.value.length > 0)
+
+    const isSelectBtnEnabled = computed(() => isLocationPicker.value || areResourcesSelected.value)
+
+    const loadDrives = async () => {
+      state.value = 'loading'
+
+      try {
+        const {
+          data: { value }
+        } = await client.value.graph.drives.listMyDrives('name')
+
+        drives.value = value.map(buildDrive)
+        spaces.value = value.filter((drive) => drive.driveType === 'project').map(buildSpace)
+        state.value = 'loaded'
+
+        if ((isInitial && props.isInitialFocusEnabled) || !isInitial) {
+          nextTick(() => focusAndAnnounceBreadcrumb(resources.value.length))
+        }
+
+        isInitial = false
+      } catch (error) {
+        console.error(error)
+
+        state.value = 'failed'
+      }
     }
-  },
 
-  created() {
-    this.loadFolder('/')
-  },
+    const loadFolder = async (path: string) => {
+      state.value = 'loading'
 
-  methods: {
-    loadFolder(path) {
-      this.state = 'loading'
+      try {
+        const { resource, children } = await webdav.value.listFiles(currentSpace, { path })
 
-      this.$client.files
-        .list(decodeURIComponent(path), 1, this.davProperties)
-        .then((resources) => {
-          resources = resources.map((resource) => buildResource(resource))
-          this.resources = resources.splice(1)
-          this.currentFolder = resources[0]
+        currentFolder.value =
+          path === '/' && currentSpace.driveType === 'personal'
+            ? { ...resource, name: proxy?.$gettext('Personal') }
+            : resource
+        resources.value = children
```

Comment:
Can you please apply the same sorting as in web? Right now there is no sorting happening afaict.

We've moved the sort helper to `web-pkg` recently. If you want to use that we can make a new release of `web-pkg`. Otherwise it's also fine if you just copy & paste the sort callback. it's not too complex. ;-)

https://github.com/owncloud/web/blob/master/packages/web-pkg/src/composables/sort/useSort.ts#L131